arangod/Cache/PlainBucket.cpp

```diff
@@ -142,7 +142,10 @@ void PlainBucket::evict(CachedValue* value, bool optimizeForInsertion) {
 
 void PlainBucket::clear() {
   TRI_ASSERT(isLocked());
-  memset(this, 0, sizeof(PlainBucket));
+  _state.clear();
+  memset(this + sizeof(BucketState), 0,
```

Comment:
What is the meaning of `this + sizeof(BucketState)`? 
I sincerely do not know... In the best case, this is the expected `(char*) this + sizeof(BucketState)`.
But in the worst case, it is `(PlainBucket*) this + sizeof(BucketState)`, and then it will write over the bounds.
Can't we change this `memset` to a simple `for... set-to-null` loop?
This would be much safer and portable, and it is likely that the compiler will still optimize it the same way.

Reply:
The type of `this` is `PlainBucket*`, so this would write _only outside_ the bounds of this instance. But why not just use two `memset` calls: `memset(&_cachedHashes, 0, sizeof(_cachedHashes))` and `memset(&_cachedData, 0, sizeof(_cachedData))`?